libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxSpeechApi.kt

```diff
@@ -73,12 +118,50 @@ class MapboxSpeechApi @JvmOverloads constructor(
      */
     fun clean(announcement: SpeechAnnouncement) {
         voiceAPI.clean(announcement)
+        VoiceInstructionsParser.parse(announcement).onValue {
+            synchronized(cachedFilesLock) {
+                val value = cachedFiles[it]
+                // when we clear fallback announcement, there is a chance we will remove the key
+                // from map and not remove the file itself
+                // since for fallback SpeechAnnouncement file is null
+                if (value?.announcement == announcement) {
+                    cachedFiles.remove(it)
+                }
+            }
+        }
+    }
+
+    internal fun predownload(instructions: List<VoiceInstructions>) {
+        mainJobController.scope.launch {
+            instructions.forEach { instruction ->
+                val typeAndAnnouncement = VoiceInstructionsParser.parse(instruction).value
+                if (typeAndAnnouncement != null && !hasTypeAndAnnouncement(typeAndAnnouncement)) {
+                    predownloadScope.launch {
+                        retrieveVoiceFile(instruction) {
+                            it.onValue { speechValue ->
+                                synchronized(cachedFilesLock) {
+                                    cachedFiles[typeAndAnnouncement] = speechValue
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    internal fun destroy() {
```

Comment:
BTW, why is it internal? maybe customers need it it too as well as we need it?

Reply:
Not really. It's only needed when predownloading is enabled. And it is enabled via other objects (`VoiceInstructionsPrefetecher`). So it probably makes sense that the prefetcher destroys `MapboxSpeechApi`, not the user.

Reply:
Probably the function should be called `cancelPredownload` or something like that.